lib/events.js

```diff
@@ -332,12 +364,11 @@ EventEmitter.prototype.removeAllListeners =
 
       if (typeof listeners === 'function') {
         this.removeListener(type, listeners);
-      } else if (Array.isArray(listeners)) {
+      } else if (listeners && listeners.length) {
         // LIFO order
-        while (listeners.length)
+        while (listeners[0])
```

Comment:
From my testing, setting `list.length = 0` and using `while (listeners.length)` is slower than setting `list[0] = undefined` and using `while (listeners[0])` for notifying that there are no more listeners.